src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Query.vb

```diff
@@ -157,9 +157,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
                 Dim propertyDefName As String = propertyDef.Name
 
-                If propertyDefName.StartsWith("$"c, StringComparison.Ordinal) AndAlso Not String.Equals(propertyDefName, StringConstants.Group, StringComparison.Ordinal) Then
+                If propertyDefName.StartsWith("$"c, StringComparison.Ordinal) AndAlso
+                   (propertyDefName.Equals(StringConstants.It) OrElse propertyDefName.Equals(StringConstants.It1) OrElse propertyDefName.Equals(StringConstants.It2)) Then
```

Comment:
Consider including an explicit `StringComparison` in `Equals()` calls.

Reply:
How do we hit the three different cases?

Reply:
> Consider including an explicit StringComparison in Equals() calls.

Added